src/bioregistry/data/bioregistry.json

```diff
@@ -32837,6 +32867,7 @@
       "wikidata": "P4229"
     },
     "name": "International Classification of Diseases, 10th Revision, Clinical Modification",
+    "pattern": "^([A-Z][0-9][0-9AB]((-[A-Z][0-9][0-9AB])|(\\.[0-9A-KXZ]([0-9A-EXYZ]([0-9A-HX][0-59A-HJKMNP-S]?)?)?)?))$",
```

Comment:
The ICD10CM pattern is based on the one at https://medium.com/@manabu.torii/regex-pattern-for-icd-10-cm-codes-5763bd66e26d by Manabu Torii. It's based on an analysis of all 2021 ICD-10-CM and therefore might be too restrictive for future releases that use additional characters for new codes.

But can possibly deal with breakage then to get the benefits of a more restrictive regex now.

Added support for code ranges like `C00-D49`

Reply:
If it it works for the 2021 and all preceding versions I have no objections.